pytket/extensions/aqt/backends/aqt_multi_zone.py

```diff
@@ -247,13 +253,34 @@ def get_result(self, handle: ResultHandle, **kwargs: KwargTypes) -> BackendResul
         """
         raise NotImplementedError
 
-    def get_compiled_circuit(
-        self, circuit: Circuit, optimisation_level: int = 2
-    ) -> Circuit:
-        raise NotImplementedError("Use get_compiled_circuit_mz for Multizone Circuits")
+    def compile_circuit_with_routing(
+        self,
+        circuit: Circuit,
+        initial_placement: Optional[ZonePlacement] = None,
+        optimisation_level: int = 2,
+    ) -> MultiZoneCircuit:
+        """
+        Compile a pytket Circuit and route it to the backend architecture
+
+        Returns a MultiZoneCircuit that conforms to the backend architecture
 
-    def get_compiled_circuit_mz(
-        self, circuit: MultiZoneCircuit, optimisation_level: int = 2
+        """
+        assert circuit.is_simple
```

Comment:
Might be better to give a more specific error here? Not sure if this could be hit by a user?